### DISCUSSION TITLE
Constrained Transport with AMR

### DISCUSSION THREAD

**Discussion by smressle** [PrincetonUniversity/athena#490] [Q&A]
Will constrained transport still result in the divergence of B being zero to machine precision when meshblocks are being created/destroyed during the simulation?

In my test problem (torus in Cartesian GR) it seems like the answer is no.

**Comment by c-white** [comment]
This could be a result of the GRUser coordinates numerically approximating areas and volumes. I will take a closer look at this.

**Reply by smressle** [reply]
I believe you may be right.  The issue is not necessarily just that the areas are approximate, it's that they are not consistently approximate.  That is, when you break down a cell face into four new smaller faces, the sum of areas will not necessarily equal the original area since the approximate area sqrt(-g) dx1 dx2 is evaluated at the face center.  Then the magnetic flux will change because of this.  

That's my guess.  So it is only a problem with GRUser coordinates and AMR when split faces into smaller ones.  I don't see an obvious solution, but I also suspect the errors introduced are small and probably not important since CT will keep the problem from multiplying.  I also suspect that HAMR has the same problem because their coordinates are quite complicated and they compute areas in the same way, though I think the magnetic fields are cell centered there so I don't know if that changes anything.

**Comment by smressle** [comment]
I should say, in general there seems to be no obvious solution.  For the specific case of cartesian KS coordinates you can just integrate everything exactly.  Easier too because sqrt(-g) =1.  You would think that because of this the numerical calculation of areas with sqrt(-g) would be nearly exact, but maybe not.

**Reply by c-white** [reply]
Yeah, the general solution is a bit tricky. As a test of the specific case here, if you overwrite `Real Determinant(const AthenaArray<Real> &g)` to just return -1, does AMR work better?

**Reply by smressle** [reply]
Trying now
